STANDARDS.md

```diff
@@ -29,6 +29,19 @@ Make sure your plugin follows the SoC design principle and has different functio
 
 Make effective use of namespacing since it will allow co-existence of ambiguous names / classes with the same name and help avoid collisions.
 
+### Recommended Code Style
+Format Java files with the Eclipse JDT formatter, using the [Spotless Gradle](https://github.com/diffplug/spotless/tree/master/plugin-gradle) plugin.
+
+Please follow these formatting guidelines:
+
+* Java indent is 4 spaces
+* Line width is 140 characters
+* Lines of code surrounded by `// tag::NAME` and `// end::NAME` comments are included in the documentation and should only be 76 characters wide not counting leading indentation. Such regions of code are not formatted automatically as it is not possible to change the line length rule of the formatter for part of a file. Please format such sections sympathetically with the rest of the code, while keeping lines to maximum length of 76 characters.
+* Wildcard imports (`import foo.bar.baz.*`) are forbidden and will cause the build to fail.
+* If *absolutely* necessary, you can disable formatting for regions of code with the `// tag::NAME` and `// end::NAME` directives, but note that these are intended for use in documentation, so please make it clear what you have done, and only do this where the benefit clearly outweighs the decrease in consistency.
+* Note that JavaDoc and block comments i.e. `/* ... */` are not formatted, but line comments i.e `// ...` are.
+* There is an implicit rule that negative boolean expressions should use the form `foo == false` instead of `!foo` for better readability of the code. While this isn't strictly enforced, if might get called out in PR reviews as something to change.
+
```

Comment:
Some plugins use checkstyle and spotless? I know we dont use checkstyle anymore. 
Would you recommend similar setup for plugins?

Reply:
Should I link the META issue here to explain the process of removal of checkstyle and enabling spotless?

Reply:
Um no, my question was slightly different. 
Would you recommend plugins to use only (and why):
- Spotless
- Spotless and Checkstyle
- Checkstyle 
- etc

Reply:
Pushed the change. Let me know what you think of it. I'll be happy to make more changes based on the suggestion. 😁

Reply:
Looks good to me. 
Probably put this section at the top(L43) L46-50 to help the readers understand why we recommend spotless for OpenSearch plugins.

Reply:
Sure. Done!